pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -207,7 +215,7 @@ func (configgen *ConfigGeneratorImpl) createGatewayHTTPFilterChainOpts(
 	services, err := env.Services() // cannot panic here because gateways do not rely on services necessarily
 	if err != nil {
 		log.Errora("Failed to get services from registry")
-		return []*filterChainOpts{}
+		return nil
```

Comment:
Metric as well - errors are not visible to users, metrics may show on dashboard.